compiler/rustc_hir_analysis/src/collect/item_bounds.rs

```diff
@@ -115,6 +115,8 @@ pub(super) fn explicit_item_bounds(
             };
             opaque_type_bounds(tcx, def_id, bounds, item_ty, *span)
         }
+        // FIXME(fmease): We should probably handle IATs here. This is only reachable when we
```

Comment:
It's not possible, and also IAT's don't have bounds... see: "bounds on `type`s in `impl`s have no effect"